rx-java/src/main/java/io/vertx/rx/java/WorkerExecutorScheduler.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2016 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.vertx.rx.java;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.WorkerExecutor;
+import io.vertx.core.impl.ConcurrentHashSet;
+import rx.Scheduler;
+import rx.Subscription;
+import rx.functions.Action0;
+import rx.plugins.RxJavaPlugins;
+import rx.plugins.RxJavaSchedulersHook;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Schedule actions for execution on a {@link io.vertx.rxjava.core.WorkerExecutor}.
+ *
+ * @author Thomas Segismont
+ */
+public class WorkerExecutorScheduler extends Scheduler {
+
+  private static final ScheduledExecutorService EXECUTOR_SERVICE = Executors.newSingleThreadScheduledExecutor(r -> {
```

Comment:
It is setup to use a thread factory which creates daemon threads. I favored
the specific executor service because the WorkerPool interface does not
have a "owner" method like the Context interface. So we could add a Vertx
parameter to the constructor, without any guarantee that it is the instance
owning the WorkerPool.

2016-08-11 1:35 GMT+02:00 testn <notifications@github.com>:

> In rx-java/src/main/java/io/vertx/rx/java/WorkerExecutorScheduler.java
> <https://github.com/vert-x3/vertx-rx/pull/46#discussion_r74350717>:
>
> > +import rx.plugins.RxJavaSchedulersHook;
> > +
> > +import java.util.concurrent.Executors;
> > +import java.util.concurrent.ScheduledExecutorService;
> > +import java.util.concurrent.ScheduledFuture;
> > +import java.util.concurrent.TimeUnit;
> > +import java.util.concurrent.atomic.AtomicBoolean;
> > +
> > +/**
> > + * Schedule actions for execution on a {@link io.vertx.rxjava.core.WorkerExecutor}.
> > + *
> > + * @author Thomas Segismont
> > + */
> > +public class WorkerExecutorScheduler extends Scheduler {
> > +
> > +  private static final ScheduledExecutorService EXECUTOR_SERVICE = Executors.newSingleThreadScheduledExecutor(r -> {
>
> not a great idea to use static ExecutorService. There is no good way to
> ensure that it is shutdown. Maybe you should rely on Vertx event loop timer
> instead?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/vert-x3/vertx-rx/pull/46/files/e7488c91f845d8c98eaad4b9bb16839546e7b18a#r74350717>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABbltpORwcACQ_82p0ECzrj-TTKpItF4ks5qemBUgaJpZM4JhJfH>
> .
>